plasmapy/utils/tests/collections_helpers_tests.py

```diff
@@ -0,0 +1,100 @@
+import unittest
+from collections import defaultdict
+from plasmapy.utils import compare_default_dicts
+
+
+class TestDictCompare(unittest.TestCase):
+    """
+    Tests custom comparing of pythons defaultdicts.
+    Note that default value of int = 0 and default of
+    str = ''
+    """
+
+    def test_same_keys(self):
+        msg = "Two equal defaultdicts should be same."
+        a = defaultdict(int, a=1, b=1, c=1)
+        b = defaultdict(int, a=1, b=1, c=1)
+        result_a_first = compare_default_dicts(a, b)
+        result_b_first = compare_default_dicts(b, a)
+        self.assertEqual(result_a_first, True, msg)
+        self.assertEqual(result_b_first, True, msg)
+
+    def test_different_values_on_intersecting_keys_same_default(self):
+        msg = "Same key, same default, but different value should fail."
+        a = defaultdict(int, a=2, b=2, c=2)
+        b = defaultdict(int, a=1, b=1, c=1)
+        result_a_first = compare_default_dicts(a, b)
+        result_b_first = compare_default_dicts(b, a)
+        self.assertEqual(result_a_first, False, msg)
+        self.assertEqual(result_b_first, False, msg)
+
+    @unittest.skip("Awaiting specific requirements")
+    def test_disjoint_values_match_default_of_other(self):
+        msg = "???"
+        a = defaultdict(str, a=0, b=0, c="")
+        b = defaultdict(int, c="", d="", e="")
```

Comment:
I would expect this to fail because `0 == ""` is `False`. @namurphy ?

Reply:
Well, both of the above boils down to whether or not we care about the default value in itself or just the k,v pairs actually present in the dicts. But they do make for some interesting edge cases!